src/firefly/js/visualize/ui/CatalogTableListField.jsx

```diff
@@ -0,0 +1,227 @@
+     * @param {string} catName the catalog name to get DD fetched from
+     */
+    updateTable(catName = 'wise_allwise_p3as_psd') {
+        this.getDD(catName);//'wise_allwise_p3as_psd'
+    }
+
+    /**
+     * Getting dd info from catalog name catName
+     * @param catName string name of the catalog for searching DD information
+     */
+    getDD(catName) {
+
+
+        const request = {id: 'GatorDD', 'catalog': catName}; //Fetch DD master table
+        TblUtil.doFetchTable(request).then((tableModel) => {
+
+            var data = tableModel.tableData.data;
+            const html = data.map((c) => {
+                return c[0] + ':' + c[1] + ' [' + c[2] + ']';
+            });
+            const cols = tableModel.tableData.columns;
+            // What to do with that?
+            // this.setState({...state},{})
+
+        }).catch((reason) => {
+                console.error(reason);
+            }
+        );
+    }
+
+    componentWillReceiveProps(np) {
+        const newFirstCatalogValue = get(np, 'data[0].value', '');
+        const oldFirstValueCatalog = get(this.props, 'data[0].value', '');
+        if (newFirstCatalogValue != oldFirstValueCatalog) {
+            this.setState(...this.state, {idx: 0});
+        } else {
+            this.setState(...this.state, {idx: np.indexClicked});
+        }
+
+    }
+
+    /**
+     * Only update when new catalog selected or different list item has been clicked
+     * @param np
+     * @returns {boolean}
+     */
+    shouldComponentUpdate(np) {
+        const newCatValue = get(np, 'value', '');
+        const oldCatValue = get(this.props, 'value', '');
+        //FIXME Doesn't work when user comes back from another tab
+        return newCatValue != oldCatValue;
+    }
+
+    render() {
+        var {data, cols, onClick, fieldKey} = this.props;//data {cat:...}
+        const indexClicked = get(this.state, 'idx', 0);
+
+        if (isEmpty(data)) {
+            return (
+                <div style={{position: 'relative'}}>
+                    <div className='loading-mask'/>
+                </div>
+            );
+        }
+
+        var items = data.map((o, index) => {
+
+
+            let clicked = false;
+            if (indexClicked === index) {
+                clicked = true;
+            }
+            return (
+
+                <CatalogTableItem
+                    key={index}
+                    onClick={onClick}
+                    isClicked={clicked}
+                    cols={cols}
+                    itemData={o}/>
+            );
+        });
+
+        return (
+            <div>
+                <table name={fieldKey}>
+                    <tbody>
+                    {items}
+                    </tbody>
+                </table>
+
+            </div>
+        );
+    }
+}
+
+export const CatalogTableListField = fieldGroupConnector(CatalogTableView, getProps, CatalogTableView.propTypes, null);
```

Comment:
I am impressed that you got your head around field group connector.  It is hard concept.